crates/bevy_winit/src/lib.rs

```diff
@@ -45,7 +84,23 @@ pub fn winit_runner(mut app: App) {
     );
 
     log::debug!("Entering winit event loop");
-    event_loop.run(move |event, event_loop, control_flow| {
+
+    let should_return_from_run = match app.resources.get::<WinitConfig>() {
```

Comment:
for simplicity, lets just use this:

```rust
let should_return_from_run = app
    .resources
    .get::<WinitConfig>()
    .map_or(false, |config| config.return_from_run);
```

Then we can rely on the alternate return_from_run impl to handle the "unsupported platform" case (see my other comment).

Reply:
On some level I kind of liked the user friendly fallback of just calling `run` on platforms that don't  support `run_return` but this is a power user feature anyhow, and the simpler implementation here is nice.

Thanks.